doc/sphinx/language/extensions/implicit-arguments.rst

```diff
@@ -498,6 +498,50 @@ in :ref:`canonicalstructures`; here only a simple example is given.
    Otherwise said, :token:`qualid` is canonically used to extend the field |c_i|
    into a complete structure built on |c_i|.
 
+   The following kinds of terms are supported for the fields |c_i| of :token:`qualid`:
+
+   * Constants (defined through :cmd:`Definition`, :cmd:`Inductive` and constructors thereof,
+     :cmd:`Record` and fields thereof, etc.) and section variables of an active section,
+     applied to zero or more arguments.
+   * Any :token:`@sort`.
+   * Literal functions:  `fun … => …`.
+   * Literal function types without type dependencies, i.e. implications: `… -> …`.
+   * Variables bound in :token:`qualid`.
+
+   When searching for a canonical extension of a field |x_i|, only the head symbol of
+   any existing instance's field |c_i| is considered.
+   We call this head symbol the *key* and we say ":token:`qualid` keys field |x_i| to |k|" when |c_i|'s
```

Comment:
You can use `:gdef:` to define a glossary term. But it would have to be something more complex than just "key", which is ambiguous.

Reply:
I don't know if that's necessary. Maybe coming up with good terminology that is worthy of being added to the glossary could be a task for the complete rework?

Reply:
Ping! Can we postpone finding a good glossary term to a future rewrite of the docs? If so, please mark this resolved.